apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/RLSManagement/RLSManagement.tsx

```diff
@@ -0,0 +1,401 @@
+    }
+
+    return fks
+      .filter((fk) => fk.columns && fk.columns.length > 0) // Only include FKs with columns
+      .map((fk) => ({
+        id: typeof fk.id === 'number' ? fk.id : 0,
+        constraint_name: fk.name || '',
+        source_id: typeof fk.tableId === 'number' ? fk.tableId : 0,
+        source_schema: schema.trim(),
+        source_table: tableName.trim(),
+        source_columns: fk.columns.map((col: { source: string; target: string }) =>
+          col.source.trim()
+        ),
+        target_id: 0,
+        target_schema: fk.schema.trim(),
+        target_table: fk.table.trim(),
+        target_columns: fk.columns.map((col: { source: string; target: string }) =>
+          col.target.trim()
+        ),
+        deletion_action: fk.deletionAction || 'NO ACTION',
+        update_action: fk.updateAction || 'NO ACTION',
+      }))
+  }
+
+  const handleGeneratePolicies = async () => {
+    if (!project?.ref || !tableName || columns.length === 0) {
+      toast.error('Unable to generate policies. Please ensure table name and columns are set.')
+      return
+    }
+
+    track('rls_generate_policies_clicked')
+
+    setIsGenerating(true)
+    try {
+      const trimmedTableName = tableName.trim()
+      const trimmedSchema = schema.trim()
+
+      const newTableForeignKeys = convertForeignKeysToConstraints(foreignKeyRelations)
+
+      const allForeignKeys = [
+        ...newTableForeignKeys,
+        ...(schemaForeignKeys ?? []).filter(
+          (existingFk) =>
+            !(
+              existingFk.source_schema === trimmedSchema &&
+              existingFk.source_table === trimmedTableName
+            )
+        ),
+      ]
+
+      const tableColumns = columns.map((col) => ({ name: col.name.trim() }))
+
+      const policies = await generateStartingPoliciesForTable({
+        table: { name: trimmedTableName, schema: trimmedSchema },
+        foreignKeyConstraints: allForeignKeys,
+        columns: tableColumns,
+        projectRef: project.ref,
+        connectionString: project.connectionString,
+        enableAi: includeSchemaMetadata,
+      })
+
+      if (policies.length === 0) {
+        setGenerateFailed(true)
+      } else {
+        setGenerateFailed(false)
+        onGeneratedPoliciesChange?.(policies)
+      }
+    } catch (error: any) {
+      console.error('Failed to generate policies:', error)
+      toast.error(error.message || 'Failed to generate policies')
+    } finally {
+      setIsGenerating(false)
+    }
+  }
+
+  const handleRlsToggleSuccess = (nextIsEnabled: boolean) => {
+    onChangeRlsEnabled?.(nextIsEnabled)
+    toast.success(
+      nextIsEnabled
+        ? 'Row Level Security has been enabled for this table.'
+        : 'Row Level Security has been disabled for this table.'
+    )
+  }
+
+  const handleRlsToggleError = (error: ResponseError) => {
+    toast.error(error.message ?? 'Unable to update Row Level Security for this table.')
+  }
+
+  const renderEnableRlsButton = () => {
```

Comment:
i'd opt to avoid declaring functions to render UI like this from a readability POV - could either declare a new component if the logic is a bit complex, or just declare directly in the final return statement

otherwise i'd need to jump around in the code to see what is being rendered 😅